exchanges/bybit/bybit_ws_cfutures.go

```diff
@@ -0,0 +1,802 @@
+	var errs common.Errors
+	for i := range channelsToSubscribe {
+		var sub WsFuturesReq
+		sub.Topic = subscribe
+
+		sub.Args = append(sub.Args, formatArgs(channelsToSubscribe[i].Channel, channelsToSubscribe[i].Params))
+		err := by.Websocket.Conn.SendJSONMessage(sub)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		by.Websocket.AddSuccessfulSubscriptions(channelsToSubscribe[i])
+	}
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+func formatArgs(channel string, params map[string]interface{}) string {
+	argStr := channel
+	for x := range params {
+		argStr += dot + fmt.Sprintf("%v", params[x])
+	}
+	return argStr
+}
+
+// UnsubscribeCoin sends a websocket message to stop receiving data from the channel
+func (by *Bybit) UnsubscribeCoin(channelsToUnsubscribe []stream.ChannelSubscription) error {
+	var errs common.Errors
+
+	for i := range channelsToUnsubscribe {
+		var unSub WsFuturesReq
+		unSub.Topic = unsubscribe
+
+		a, err := by.GetPairAssetType(channelsToUnsubscribe[i].Currency)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		formattedPair, err := by.FormatExchangeCurrency(channelsToUnsubscribe[i].Currency, a)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		unSub.Args = append(unSub.Args, channelsToUnsubscribe[i].Channel+dot+formattedPair.String())
+		err = by.Websocket.Conn.SendJSONMessage(unSub)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		by.Websocket.RemoveSuccessfulUnsubscriptions(channelsToUnsubscribe[i])
+	}
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+// wsCoinReadData gets and passes on websocket messages for processing
+func (by *Bybit) wsCoinReadData() {
+	by.Websocket.Wg.Add(1)
+	defer by.Websocket.Wg.Done()
+
+	for {
+		select {
+		case <-by.Websocket.ShutdownC:
+			return
+		default:
+			resp := by.Websocket.Conn.ReadMessage()
+			if resp.Raw == nil {
+				return
+			}
+
+			err := by.wsCoinHandleData(resp.Raw)
+			if err != nil {
+				by.Websocket.DataHandler <- err
+			}
+		}
+	}
+}
+
+func (by *Bybit) wsCoinHandleData(respRaw []byte) error {
+	var multiStreamData map[string]interface{}
+	err := json.Unmarshal(respRaw, &multiStreamData)
+	if err != nil {
+		return err
+	}
+
+	if t, ok := multiStreamData["topic"].(string); ok {
+		topics := strings.Split(t, dot)
+		if len(topics) < 1 {
+			return errors.New(by.Name + " - topic could not be extracted from response")
+		}
+
+		if wsType, ok := multiStreamData["type"].(string); ok {
+			switch topics[0] {
+			case wsOrder25, wsOrder200:
+				switch wsType {
+				case wsOperationSnapshot:
+					var response WsFuturesOrderbook
+					err = json.Unmarshal(respRaw, &response)
+					if err != nil {
+						return err
+					}
+
+					var p currency.Pair
+					p, err = currency.NewPairFromString(response.OBData[0].Symbol)
+					if err != nil {
+						return err
+					}
+
+					var a asset.Item
+					a, err = by.GetPairAssetType(p)
```

Comment:
There are multiple assets that contain the same currency pair string and this is giving false positives or first match across your implementations. I have added more info in _wrapper.go comment. There needs to be further differentiation.

Reply:
@shazbert You are right. I have understood your concern.
Can you please guide me on how can we solve it? I have been looking for a solution to this issue but didn't succeed for the past 2 days.

Reply:
@Jdpurohit  Spot websocket is working but others are not. So for now, we need to filter out channel subscriptions that are not spot-related. 

```
SPOT SUBSCRIBE: [{bookTicker BTC-USD coinmarginedfutures map[]} {trade BTC-USD coinmarginedfutures map[]} {depth BTC-USD coinmarginedfutures map[]} {bookTicker XRP-USD usdtmarginedfutures map[]} {trade XRP-USD usdtmarginedfutures map[]} {depth XRP-USD usdtmarginedfutures map[]} {bookTicker ETH-USD futures map[]} {trade ETH-USD futures map[]} {depth ETH-USD futures map[]} {bookTicker ETH-USDT spot map[]} {trade ETH-USDT spot map[]} {depth ETH-USDT spot map[]}]
[ERROR]|EXCHANGE| 21/03/2022 10:37:45 |Bybit Received unhandle message on websocket: map[code:-100010 desc:Invalid Symbols!]
```

Also we can probably stop using method function `GetPairAssetType()` to get asset. We already know that the bybit_websocket.go handling is spot only. And the others `_ws_futures...` can be changed to their hardcoded assets. 

In the future when we want to enable all of the assets we can have a `wsReadData()` method function with an asset and a websocket connection as parameters, and run separate routines for each.

Reply:
Thank @shazbert. Got it. I have made the required changes.